sdk/confidentialledger/confidential-ledger-rest/src/confidentialLedger.ts

```diff
@@ -0,0 +1,48 @@
+import { ClientOptions } from "@azure-rest/core-client";
+import { TokenCredential, CertificateCredential, isCertificateCredential } from "@azure/core-auth";
+import { PipelinePolicy } from "@azure/core-rest-pipeline";
+import { Agent } from "./agent";
+import GeneratedConfidentialLedger, {
+  ConfidentialLedgerRestClient,
+} from "./generated/src/confidentialLedger";
+
+export default function ConfidentialLedger(
+  ledgerBaseUrl: string,
+  ledgerTlsCertificate: string,
+  credentials: TokenCredential | CertificateCredential,
+  options?: ClientOptions
+): ConfidentialLedgerRestClient {
+  const confidentialLedger = GeneratedConfidentialLedger(
+    ledgerBaseUrl,
+    credentials as any,
+    options
+  );
+
+  confidentialLedger.pipeline.addPolicy(getCertValidationPolicy(ledgerTlsCertificate, credentials));
+
+  return confidentialLedger;
+}
+
+function getCertValidationPolicy(
+  ledgerTlsCertificate: string,
+  credential: TokenCredential | CertificateCredential
+): PipelinePolicy {
+  return {
+    name: "ledgerTlsCertificatePolicy",
+    sendRequest: (request, next) => {
+      // Create default agent and options if they don't exist
+      request.agent = request.agent ?? new Agent();
+      request.agent.options = request.agent.options ?? {};
```

Comment:
I'm confused by this, aren't options passed to the Agent constructor instead of dangling off the instance?

Reply:
Yeah, initially I was just doing

```typescript
request.agent = new Agent({ca: "", cert: "", key: ""})
```

The problem I saw with this, is that if another policy sets a custom Agent, they will conflict and one of them will be dropped. That's the reason I switched to exposing the options.

However I don't have strong feelings about this, it may be optimizing for the edge case. I would be fine if we think this would not be a common issue. It is always easier to expose the options in the future if we find it is an issue.

Reply:
I'm confused about the scenario here. If options are hanging off of `request.agent`, and some policy reassigns `request.agent` to something else, the options will still get lost unless that policy is super considerate and tries to look at the existing options, which I think is assuming too much generosity on the part of the conflicting policy. 😄

Reply:
Yeah that's a fair point!